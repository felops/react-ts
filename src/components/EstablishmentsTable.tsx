import { Link } from "react-router";
import { Establishment } from "../api/ratingsAPI";

const headerStyle: { [key: string]: string | number } = {
  paddingBottom: "10px",
  textAlign: "left",
  fontSize: "20px",
};

interface EstablishmentsTableProps {
  establishments: Establishment[] | undefined;
  isLoading?: boolean;
}

export const EstablishmentsTable = ({
  establishments,
  isLoading
}: EstablishmentsTableProps) => { 
  return (
    <>
      <table>
        <thead>
          <tr>
            <th style={headerStyle}>Business Name</th>
            <th style={headerStyle}>Rating Value</th>
          </tr>
        </thead>
        <tbody>
          {establishments && establishments?.map(establishment => (
            <tr key={establishment?.FHRSID} style={{ fontSize: "20px" }}>
              <td>
                <Link to={`${establishment.FHRSID}/details`}>{establishment?.BusinessName}</Link>
              </td>
              <td>{establishment?.RatingValue}</td>
            </tr>
          ))}
        </tbody>
      </table>
      {isLoading && <div>Loading...</div>}
    </>
  );
};
