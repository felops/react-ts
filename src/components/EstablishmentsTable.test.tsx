import {render, screen} from '@testing-library/react' 
import { EstablishmentsTable } from './EstablishmentsTable';

const mockData = [
  {
    BusinessName: "Joe",
    RatingValue: "5"
  },
  {
    BusinessName: "Habib",
    RatingValue: "4"
  },
  {
    BusinessName: "Mc Hogwarts",
    RatingValue: "3"
  },
]

describe("EstablishmentsTable", () => {
  it('renders correctly', () => {
    const { asFragment } = render(<EstablishmentsTable establishments={mockData} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('has head titles', () => {
    render(<EstablishmentsTable establishments={mockData} />);
    const headTitles = screen.getAllByRole('columnheader').map(th => th.textContent);
    
    expect(headTitles).toEqual(['Business Name', 'Rating Value']);
  });

  it('render all data', async () => {
    render(<EstablishmentsTable establishments={mockData} />);
    const allLines = await screen.findAllByRole('row');

    expect(allLines.length - 1).toEqual(mockData.length); // minus 1 for the header row
  });

  it('render body rows with correct style', async () => {
    render(<EstablishmentsTable establishments={mockData} />);
    const allBodyLines = await screen.findAllByRole('row');
    expect(allBodyLines[1].style.fontSize).toEqual("20px");
  });
});
