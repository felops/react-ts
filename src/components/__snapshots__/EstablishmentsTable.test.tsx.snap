// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EstablishmentsTable renders correctly 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th
          style="padding-bottom: 10px; text-align: left; font-size: 20px;"
        >
          Business Name
        </th>
        <th
          style="padding-bottom: 10px; text-align: left; font-size: 20px;"
        >
          Rating Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        style="font-size: 20px;"
      >
        <td>
          <a
            data-discover="true"
            href="/22999/details"
          >
            Joe
          </a>
        </td>
        <td>
          5
        </td>
      </tr>
      <tr
        style="font-size: 20px;"
      >
        <td>
          <a
            data-discover="true"
            href="/67890/details"
          >
            Habib
          </a>
        </td>
        <td>
          4
        </td>
      </tr>
      <tr
        style="font-size: 20px;"
      >
        <td>
          <a
            data-discover="true"
            href="/09888/details"
          >
            Mc Hogwarts
          </a>
        </td>
        <td>
          3
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
